sdks/python/setup.py

```diff
@@ -378,7 +378,6 @@ def get_portability_package_data():
           'numpy>=1.14.3,<2.3.0',  # Update pyproject.toml as well.
           'objsize>=0.6.1,<0.8.0',
           'packaging>=22.0',
-          'pymongo>=3.8.0,<5.0.0',
```

Comment:
I like this change, thanks. Up to you on whether you want to take this on, but I think we would benefit from doing this with more dependencies. Specifically, the following are used very lightly and are not needed for core functionality:

- hdfs (would need a new extra)
- jsonschema (just used in yaml, could be moved to that extra)
- pydot (just used in interactive, could be moved to that extra)
- redis

> I also feel that we should update the [CHANGES.md](https://github.com/apache/beam/blob/master/CHANGES.md) file to mention this change.

It is breaking, so I agree - could you please do this as part of this PR?

Reply:
Thank you for highlighting others. I was just thinking about Redis. I can take care of them in this changeset.